### PR TITLE
Bugfix/issue 2412 examples implementation

### PR DESCRIPTION
**Pull request by ralphdoe**
Examples annotation and examples object.
#2412

### CODE REVIEW

File: modules/swagger-annotations/src/main/java/io/swagger/oas/annotations/Parameter.java [hunk 1]
```diff
@@ -113,12 +114,12 @@
    * @return whether or not this parameter is hidden
    */
   boolean hidden() default false;
-  
+
   /**
-   * Provides an array examples of the schema.  When associated with a specific media type, the example string shall be parsed by the consumer to be treated as an object or an array.  Ignored if the properties content or array are specified.
-   * @return the list of examples for this parameter
+   * An array of examples used to show the use of the associated schema.
```
Comment (frantuma) [inline modules/swagger-annotations/src/main/java/io/swagger/oas/annotations/Parameter.java]: change to "An array of examples of the schema"

File: modules/swagger-annotations/src/main/java/io/swagger/oas/annotations/Parameter.java [hunk 2]
```diff
@@ -113,12 +114,12 @@
...
+   * @return the list of examples
```
Comment (frantuma) [inline modules/swagger-annotations/src/main/java/io/swagger/oas/annotations/Parameter.java]: change to @return array of examples of the parameter

File: modules/swagger-jaxrs2/src/main/java/io/swagger/jaxrs2/OperationParser.java [hunk 3]
```diff
@@ -78,6 +78,10 @@
             parameterObject.setIn(parameter.in());
             isEmpty = false;
         }
+        if (StringUtils.isNotBlank(parameter.example())) {
+            parameterObject.setIn(parameter.example());
```
Comment (frantuma) [inline modules/swagger-jaxrs2/src/main/java/io/swagger/jaxrs2/OperationParser.java]: I guess this is wrong, would be `parameterObject.setParameter` instead? also it means no tests current cover the case, and should be added

File: modules/swagger-jaxrs2/src/main/java/io/swagger/jaxrs2/OperationParser.java [hunk 4]
```diff
@@ -422,35 +436,6 @@ public static void applyProduces(Produces classProduces, Produces methodProduces
         return Optional.empty();
     }
 
-    public static Optional<MediaType> getMediaType(MediaType mediaType, ExampleObject example) {
```
Comment (frantuma) [inline modules/swagger-jaxrs2/src/main/java/io/swagger/jaxrs2/OperationParser.java]: remove related unused imports IOException and Json

File: modules/swagger-jaxrs2/src/test/java/io/swagger/jaxrs2/resources/PetResource.java [hunk 5]
```diff
@@ -21,6 +21,7 @@
 import io.swagger.oas.annotations.Operation;
 import io.swagger.oas.annotations.Parameter;
 import io.swagger.oas.annotations.media.Content;
+import io.swagger.oas.annotations.media.ExampleObject;
```
Comment (frantuma) [inline modules/swagger-jaxrs2/src/test/java/io/swagger/jaxrs2/resources/PetResource.java]: unused import

File: modules/swagger-annotations/src/main/java/io/swagger/oas/annotations/Parameter.java (hunk 1 above)
Comment (ralphdoe) [reply modules/swagger-annotations/src/main/java/io/swagger/oas/annotations/Parameter.java]: Changed!

File: modules/swagger-annotations/src/main/java/io/swagger/oas/annotations/Parameter.java (hunk 2 above)
Comment (ralphdoe) [reply modules/swagger-annotations/src/main/java/io/swagger/oas/annotations/Parameter.java]: Changed!

File: modules/swagger-jaxrs2/src/test/java/io/swagger/jaxrs2/resources/PetResource.java (hunk 5 above)
Comment (ralphdoe) [reply modules/swagger-jaxrs2/src/test/java/io/swagger/jaxrs2/resources/PetResource.java]: Removed!

File: modules/swagger-jaxrs2/src/main/java/io/swagger/jaxrs2/OperationParser.java (hunk 4 above)
Comment (ralphdoe) [reply modules/swagger-jaxrs2/src/main/java/io/swagger/jaxrs2/OperationParser.java]: Removed!

File: modules/swagger-jaxrs2/src/main/java/io/swagger/jaxrs2/OperationParser.java (hunk 3 above)
Comment (ralphdoe) [reply modules/swagger-jaxrs2/src/main/java/io/swagger/jaxrs2/OperationParser.java]: Replaced by parameterObject.setExample and test added.

**Comment by frantuma** [comment]
@ralphdoe examples related code looks good, the one thing is that this branch/pr seems to also include commits from #2404 which is still in discussion, not sure why you added them here also; are these changes needed for the example related stuff? if so we must wait for finalization of #2404, otherwise can you remove those, so to keep here only stuff related to examples?

**Pull request merged**